File: src/test/fixtures/go/service.go
package main

import "fmt"

func CreateUser(name string, email string) UserProfile {
	user := UserProfile{}
	user.Name = name
	user.Email = email
	return user
}

func GetCompanyOwner(company CompanyInfo) UserProfile {
	return company.Owner
}

func ProcessEntity(entity BaseEntity) {
	fmt.Println(entity.ID)
}

func GetTimestamped(id int) TimestampedEntity {
	return TimestampedEntity{}
}

File: src/test/fixtures/csharp/Service.cs
namespace Fixtures
{
    public class Service
    {
        public UserProfile CreateUser(string name, string email)
        {
            var user = new UserProfile();
            user.Name = name;
            user.Email = email;
            return user;
        }

        public UserProfile GetCompanyOwner(CompanyInfo company)
        {
            return company.GetOwner();
        }

        public void ProcessEntity(BaseEntity entity)
        {
            Console.WriteLine(entity.Id);
        }

        public TimestampedEntity GetTimestamped(int id)
        {
            return new TimestampedEntity();
        }
    }
}
